src/posix/platform/openthread-posix-config.h

```diff
@@ -137,6 +137,38 @@
 #define OPENTHREAD_POSIX_CONFIG_NETIF_PREFIX_ROUTE_METRIC 0
 #endif
 
+/**
+ * @def OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE
+ *
+ * Define as 1 to add OMR routes to POSIX kernel when OMR prefixes are changed in netdata.
+ *
+ */
+#ifndef OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE
+#define OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+#endif
+
+/**
+ * @def OPENTHREAD_POSIX_CONFIG_OMR_ROUTE_PRIORITY
+ *
+ * This macro defines the priority of OMR routes added to kernel. The larger the number, the lower the priority. We
+ * need to assign a high priority to such routes so that kernel prefers the Thread link rather than infrastructure.
+ * Otherwise we may unnecessarily transmit packets via infrastructure, which potentially causes looping issue.
+ *
+ */
+#ifndef OPENTHREAD_POSIX_CONFIG_OMR_ROUTE_PRIORITY
+#define OPENTHREAD_POSIX_CONFIG_OMR_ROUTE_PRIORITY 1
+#endif
+
+/**
+ * @def OPENTHREAD_POSIX_CONFIG_MAX_OMR_ROUTE_NUM
+ *
+ * This macro defines the max number of OMR routes that can be added to kernel.
+ *
+ */
+#ifndef OPENTHREAD_POSIX_CONFIG_MAX_OMR_ROUTE_NUM
+#define OPENTHREAD_POSIX_CONFIG_MAX_OMR_ROUTE_NUM 8
```

Comment:
> For `OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE` option, may `#define OPENTHREAD_POSIX_CONFIG_MAX_OMR_ROUTE_NUM OPENTHREAD_CONFIG_IP6_SLAAC_NUM_ADDRESSES` for consistency? 

That's a good point. Will change. 

> If so, another thought is may consider adding/removing omr route when per omr slaac address is added or removed, other than acting per `OT_CHANGED_THREAD_NETDATA`?

It adds complexity but seems not that useful. Even if there are 5 OMR addresses (which is rare) in NetData, and only 4 OMR SLAAC address was successfully, it's still fine to add 5 OMR routes. These routes can still be used to forward messages to the Thread network.

Thoughts?

Reply:
Changed to use OPENTHREAD_CONFIG_IP6_SLAAC_NUM_ADDRESSES by default.

Reply:
> > For `OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE` option, may `#define OPENTHREAD_POSIX_CONFIG_MAX_OMR_ROUTE_NUM OPENTHREAD_CONFIG_IP6_SLAAC_NUM_ADDRESSES` for consistency?
> 
> That's a good point. Will change.
> 
> > If so, another thought is may consider adding/removing omr route when per omr slaac address is added or removed, other than acting per `OT_CHANGED_THREAD_NETDATA`?
> 
> It adds complexity but seems not that useful. Even if there are 5 OMR addresses (which is rare) in NetData, and only 4 OMR SLAAC address was successfully, it's still fine to add 5 OMR routes. These routes can still be used to forward messages to the Thread network.

Generally there would be 3 OMR prefixes in maximum when stable, meanwhile I agree that there might be cases that more than `OPENTHREAD_CONFIG_NETDATA_PUBLISHER_DESIRED_NUM_ON_MESH_PREFIX_ENTRIES` OMR prefixes available in some really rare cases.
https://github.com/openthread/openthread/blob/2e625bfe39768e0fdcfd5a40f9fbc49673b30686/src/core/config/netdata_publisher.h#L130


The reasons for my proposal to  `adding/removing omr route when per omr slaac address is added or removed` are that both `OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE` and `OPENTHREAD_POSIX_CONFIG_NETIF_PREFIX_ROUTE_METRIC` are introduced to serve the same purpose

1) nice to have the same trigger source 
  `OT_CHANGED_THREAD_NETDATA` event may be triggered more frequent than `UpdateUnicastLinux`, e.g. when extra border router sub tlv, or other external routes/services/servers are registered etc, it might be not necessary to perform the multiple loops in `UpdateOmrRoutes`.

2) nice to have similar behavior/effect for both options under  the rare scenarios specified above

Reply:
Even though the two features are both intended to be used to add high-priority OMR routes for us, they are indeed different features. 
`OPENTHREAD_POSIX_CONFIG_INSTALL_OMR_ROUTES_ENABLE`, as the name suggests, adds kernel routes for OMR prefixes. So, it should be natural to implement it by responding to network data changes since network data is the source of truth for OMR prefixes. So, I am inclined to keep the current implementation. 

> `OT_CHANGED_THREAD_NETDATA` event may be triggered more frequent than `UpdateUnicastLinux`

It should be fine to check the OMR addresses and kernel routes more than necessary. We won't modify kernel routes if not necessary.

Reply:
Prefer to keep the code as is since it's straightforward to install OMR routes according to network data changes.